### ISSUE TITLE
[Support]: frigate.comms.mqtt: Unable to publish to stats: client is not connected

### ISSUE THREAD

**Issue by sgzmd** [blakeblackshear/frigate#8083]
### Describe the problem you are having

I am running Frigate in Docker Compose, alongside Mosquitto and go2rpc. Frigate refuses to connect to the MQTT with the following error message in logs:

```
2023-10-06 16:18:46.708048635  [2023-10-06 16:18:46] frigate.comms.mqtt             ERROR   : Unable to publish to Outdoor/all: client is not connected
2023-10-06 16:19:27.333091671  [2023-10-06 16:19:27] frigate.comms.mqtt             ERROR   : Unable to publish to Outdoor/motion: client is not connected
2023-10-06 16:19:58.057959711  [2023-10-06 16:19:58] frigate.comms.mqtt             ERROR   : Unable to publish to stats: client is not connected
```

I have verified that I can indeed connect from Frigate container to Mosquitto and publish messages there like this:

```
root@bi2:/opt/frigate# mosquitto_pub -h myhostname  -p 1883 -t Hello -m World
```

I tried replacing `myhostname` in Frigate config to IP address of the host, IP address of the container as well as `127.0.0.1`. I have also tried to run another MQTT server not in docker (neither require authentication and both allow anonymous posters). At this point I'm at my wit's end as per what to do. 

### Version

0.12.1-367D724

### Frigate config file

```yaml
mqtt:
  host: myhostname
  port: 1883
logger:
  default: info
  logs:
    frigate.mqtt: debug
cameras:
  Outdoor:
    ffmpeg:
      hwaccel_args: preset-vaapi
      inputs:
        - path: rtsp://myhostname:18554/Outdoor
          input_args: preset-rtsp-restream
          roles:
            - detect
            - record
    record:
      enabled: True
      events:
        retain:
          default: 100
          mode: motion
    detect:
      enabled: True
      width: 640
      height: 480
      fps: 10

detectors:
  coral:
    type: edgetpu
    device: usb  

ffmpeg:
  hwaccel_args:
    - -hwaccel
    - vaapi
    - -hwaccel_device
    - /dev/dri/renderD128
    - -hwaccel_output_format
    - yuv420p
```

### Relevant log output

```shell
2023-10-06 16:18:46.708048635  [2023-10-06 16:18:46] frigate.comms.mqtt             ERROR   : Unable to publish to Outdoor/all: client is not connected
2023-10-06 16:19:27.333091671  [2023-10-06 16:19:27] frigate.comms.mqtt             ERROR   : Unable to publish to Outdoor/motion: client is not connected
2023-10-06 16:19:58.057959711  [2023-10-06 16:19:58] frigate.comms.mqtt             ERROR   : Unable to publish to stats: client is not connected
```

### FFprobe output from your camera

```shell
N/A
```

### Frigate stats

```json
{"Outdoor":{"camera_fps":10.0,"capture_pid":306,"detection_enabled":1,"detection_fps":0.0,"ffmpeg_pid":310,"pid":304,"process_fps":10.0,"skipped_fps":0.0},"cpu_usages":{"%Cpu(s):":{"cpu":"id,","mem":"0.2"},"1":{"cpu":"0.0","mem":"0.0"},"109":{"cpu":"0.0","mem":"0.0"},"112":{"cpu":"1.3","mem":"3.1"},"116":{"cpu":"0.0","mem":"0.2"},"134":{"cpu":"0.0","mem":"0.1"},"135":{"cpu":"0.0","mem":"0.1"},"136":{"cpu":"0.0","mem":"0.1"},"137":{"cpu":"0.0","mem":"0.1"},"15":{"cpu":"0.0","mem":"0.0"},"16":{"cpu":"0.0","mem":"0.0"},"24":{"cpu":"0.0","mem":"0.0"},"25":{"cpu":"0.0","mem":"0.0"},"26":{"cpu":"0.0","mem":"0.0"},"27":{"cpu":"0.0","mem":"0.0"},"28":{"cpu":"0.0","mem":"0.0"},"281":{"cpu":"0.0","mem":"1.0"},"29":{"cpu":"0.0","mem":"0.0"},"295":{"cpu":"0.3","mem":"0.1"},"296":{"cpu":"0.0","mem":"1.3"},"298":{"cpu":"1.3","mem":"1.1"},"30":{"cpu":"0.0","mem":"0.0"},"304":{"cpu":"2.3","mem":"1.2"},"305":{"cpu":"0.0","mem":"0.2"},"306":{"cpu":"3.0","mem":"1.1"},"31":{"cpu":"0.0","mem":"0.0"},"310":{"cpu":"8.0","mem":"0.9"},"312":{"cpu":"0.0","mem":"0.2"},"32":{"cpu":"0.0","mem":"0.0"},"41":{"cpu":"0.0","mem":"0.0"},"42":{"cpu":"0.3","mem":"0.0"},"4861":{"cpu":"0.0","mem":"0.0"},"4934":{"cpu":"0.0","mem":"0.0"},"80":{"cpu":"0.0","mem":"0.0"},"81":{"cpu":"0.0","mem":"0.0"},"82":{"cpu":"0.0","mem":"0.0"},"847":{"cpu":"0.0","mem":"0.0"},"MiB":{"cpu":"5589.9","mem":"avail"},"PID":{"cpu":"%CPU","mem":"%MEM"},"Tasks:":{"cpu":"stopped,","mem":"0"},"top":{"cpu":"users,","mem":"load"}},"detection_fps":0.0,"detectors":{"coral":{"detection_start":0.0,"inference_speed":11.74,"pid":296}},"gpu_usages":{"intel-vaapi":{"gpu":"5.84 %","mem":"- %"}},"service":{"last_updated":1696606083,"latest_version":"0.12.1","storage":{"/dev/shm":{"free":65.8,"mount_type":"tmpfs","total":67.1,"used":1.3},"/media/frigate/clips":{"free":1958742.8,"mount_type":"fuseblk","total":8001545.0,"used":6042802.3},"/media/frigate/recordings":{"free":1958742.8,"mount_type":"fuseblk","total":8001545.0,"used":6042802.3},"/tmp/cache":{"free":995.5,"mount_type":"tmpfs","total":1000.0,"used":4.5}},"temperatures":{},"uptime":559,"version":"0.12.1-367d724"}}
```

### Operating system

Debian

### Install method

Docker Compose

### Coral version

USB

### Network connection

Wired

### Camera make and model

Reolink

### Any other information that may be helpful

Recreating Docker container doesn't help

**Comment by sgzmd** [comment]
So through some hacking of Docker I got it to the point that Frigate now _is_ connecting to Mosquitto. In the logs I see:

```
frigate           | 2023-10-07 14:32:20.136140332  [2023-10-07 14:32:20] frigate.comms.mqtt             ERROR   : Unable to publish to Outdoor/person/snapshot: client is not connected
mqtt              | 1696685547: New connection from 172.26.0.3:48473 on port 1883.
mqtt              | 1696685547: Client <unknown> disconnected due to protocol error.
frigate           | 2023-10-07 14:32:38.221267741  [2023-10-07 14:32:38] frigate.comms.mqtt             ERROR   : Unable to publish to stats: client is not connected
```

**Comment by NickM-27** [comment]
something still seems weird with having that protocol error

**Comment by sgzmd** [comment]
Indeed. So far I've tried to configure password on Mosquitto, and make 
Frigate use it, which didn't have any desired effect. I can connect to 
Mosquitto using this username/password using other clients.

Roman ***@***.*** <https://twitter.com/sgzmd>" Kirillov


------ Original Message ------
From "Nicolas Mowen" ***@***.***>
To "blakeblackshear/frigate" ***@***.***>
Cc "sgzmd" ***@***.***>; "Author" ***@***.***>
Date 07/10/2023 14:34:47
Subject Re: [blakeblackshear/frigate] [Support]: frigate.comms.mqtt: 
Unable to publish to stats: client is not connected (Issue #8083)

**Comment by NickM-27** [comment]
may be something weird with the network still, mosquitto is very common to use with frigate and I also use it myself

**Comment by sgzmd** [comment]
I have no doubts about it. Is there some way I can increase the 
verbosity of logging in Frigate so it starts spewing out all of the 
errors and exceptions as-is?  I have

```
logger:
   default: info
   logs:
     frigate.mqtt: debug
```

this set up.


Roman ***@***.*** <https://twitter.com/sgzmd>" Kirillov


------ Original Message ------
From "Nicolas Mowen" ***@***.***>
To "blakeblackshear/frigate" ***@***.***>
Cc "sgzmd" ***@***.***>; "Author" ***@***.***>
Date 07/10/2023 14:44:10
Subject Re: [blakeblackshear/frigate] [Support]: frigate.comms.mqtt: 
Unable to publish to stats: client is not connected (Issue #8083)

**Comment by NickM-27** [comment]
it would be `frigate.comms.mqtt: debug`

**Comment by sgzmd** [comment]
So I enabled debugging globally. What is really curious, I am not seeing any debug messages from this branch of code:

```python
    def _on_connect(
        self,
        client: mqtt.Client,
        userdata: Any,
        flags: Any,
        rc: mqtt.ReasonCodes,
    ) -> None:
        """Mqtt connection callback."""
        threading.current_thread().name = "mqtt"
        if rc != 0:
            if rc == 3:
                logger.error(
                    "Unable to connect to MQTT server: MQTT Server unavailable"
                )
            elif rc == 4:
                logger.error(
                    "Unable to connect to MQTT server: MQTT Bad username or password"
                )
            elif rc == 5:
                logger.error("Unable to connect to MQTT server: MQTT Not authorized")
            else:
                logger.error(
                    "Unable to connect to MQTT server: Connection refused. Error code: "
                    + str(rc)
                )

        self.connected = True
        logger.debug("MQTT connected")
        client.subscribe(f"{self.mqtt_config.topic_prefix}/#")
        self._set_initial_topics()
```

As if it is never called

**Comment by NickM-27** [comment]
right, meaning it is never connecting to mosquitto (which makes sense because generally there are no logs in mosquitto)

**Comment by sgzmd** [comment]
I got it. 

```python
        if self.mqtt_config.tls_ca_certs is not None:
            if (
                self.mqtt_config.tls_client_cert is not None
                and self.mqtt_config.tls_client_key is not None
            ):
                self.client.tls_set(
                    self.mqtt_config.tls_ca_certs,
                    self.mqtt_config.tls_client_cert,
                    self.mqtt_config.tls_client_key,
                )
            else:
                self.client.tls_set(self.mqtt_config.tls_ca_certs)
```

I had tls certs specified in the config, although they weren't used. When I commented that line, things started magically working. What is curious are two things:

1. Setting TLS on connection that doesn't require it in theory shouldn't break things
2. Even if it should, at the very least there should be some sensible output in the logs.

**Comment by stokedout** [comment]
I've experienced and fixed this for those using Mosquito via docker compose. Initially I seen this partial log upon running `docker compose up`

```
x@x:~/frigate$ docker compose up
[+] Running 4/3
 ✔ Network frigate_default                                              Created                                                                                                                        0.1s 
 ✔ Container frigate                                                    Created                                                                                                                        0.2s 
 ✔ Container mosquitto                                                  Created                                                                                                                        0.2s 
 ! mosquitto Published ports are discarded when using host network mode                                                                                                                                0.0s 
Attaching to frigate, mosquitto
mosquitto  | 1710016651: mosquitto version 2.0.18 starting
mosquitto  | 1710016651: Config loaded from /mosquitto/config/mosquitto.conf.
mosquitto  | 1710016651: Starting in local only mode. Connections will only be possible from clients running on this machine.
mosquitto  | 1710016651: Create a configuration file which defines a listener to allow remote access.
mosquitto  | 1710016651: For more details see https://mosquitto.org/documentation/authentication-methods/
mosquitto  | 1710016651: Opening ipv4 listen socket on port 1883.
mosquitto  | 1710016651: Opening ipv6 listen socket on port 1883.
mosquitto  | 1710016651: mosquitto version 2.0.18 running
frigate    | s6-rc: info: service s6rc-fdholder: starting
frigate    | s6-rc: info: service s6rc-oneshot-runner: starting
frigate    | s6-rc: info: service s6rc-oneshot-runner successfully started
```

It was not picking up the custom mosquitto config I had created until I changed the docker compose to:

```yaml
version: "3.9"
services:
  frigate:
    container_name: frigate
    restart: unless-stopped
    image: ghcr.io/blakeblackshear/frigate:stable
    volumes:
      - ./config:/config
      - ./storage:/media/frigate
      - type: tmpfs # Optional: 1GB of memory, reduces SSD/SD Card wear
        target: /tmp/cache
        tmpfs:
          size: 1000000000
    ports:
      - 5000:5000
      - 8554:8554
  mosquitto:
    image: eclipse-mosquitto
    container_name: mosquitto
    restart: unless-stopped
    network_mode: host
    volumes:
      - ./mosquitto/config:/mosquitto/config
      - ./mosquitto/data:/mosquitto/data
      - ./mosquitto/log:/mosquitto/log
    ports:
      - 1883:1883
      - 9001:9001
```


The volume `- ./mosquitto/config:/mosquitto/config` overrides the default config values. 

**./mosquitto/config/mosquitto.conf**
```yaml
persistence true
persistence_location /mosquitto/data/
log_dest file /mosquitto/log/mosquitto.log

listener 1883
allow_anonymous false
password_file /mosquitto/config/mosquitto.pwd
```

**. /mosquitto/config/mosquitto.pwd**
```yaml
myusername:$7$101$Mencrypted gobble dee gook password==
```

**Comment by nerdosity** [comment]
It's happening the same to me.
No log details, even with mqtt debug enabled.

Relevant config:

`mqtt:
  enabled: true
  host: 192.168.0.1
  port: 1883
#  topic_prefix: frigate
#  client_id: frigate
  user: homeassistant
  password: REDACTED (same for every MQTT enabled device)
  stats_interval: 60
#  tls_ca_certs:
  # Optional: tls_client_cert and tls_client key in order to use self-signed client
  # certificates (default: None)
  # NOTE: certificate must not be password-protected
  #       do not set user and password when using a client certificate
#  tls_client_cert:
#  tls_client_key:
  # Optional: tls_insecure (true/false) for enabling TLS verification of
  # the server hostname in the server certificate (default: None)
#  tls_insecure:

logger:
  default: info
  logs:
    frigate.comms.mqtt: debug`

`2024-04-24 14:40:24.395746103  [INFO] Preparing Frigate...
2024-04-24 14:40:24.612619677  [INFO] Starting Frigate...
2024-04-24 14:40:27.375142851  [2024-04-24 14:40:27] frigate.app                    INFO    : Starting Frigate (0.13.2-6476f8a)
2024-04-24 14:40:32.576751036  [2024-04-24 14:40:32] peewee_migrate.logs            INFO    : Starting migrations
2024-04-24 14:40:32.581112665  [2024-04-24 14:40:32] peewee_migrate.logs            INFO    : There is nothing to migrate
2024-04-24 14:40:32.594340669  [2024-04-24 14:40:32] frigate.app                    INFO    : Recording process started: 459
2024-04-24 14:40:32.596854967  [2024-04-24 14:40:32] frigate.app                    INFO    : go2rtc process pid: 89
2024-04-24 14:40:32.615100126  [2024-04-24 14:40:32] detector.coral_pci             INFO    : Starting detection process: 469
2024-04-24 14:40:32.632916695  [2024-04-24 14:40:32] frigate.app                    INFO    : Output process started: 471
2024-04-24 14:40:32.640134541  [2024-04-24 14:40:32] frigate.detectors.plugins.edgetpu_tfl INFO    : Attempting to load TPU as pci
2024-04-24 14:40:32.644727828  [2024-04-24 14:40:32] frigate.detectors.plugins.edgetpu_tfl INFO    : TPU found
2024-04-24 14:40:32.654141696  [2024-04-24 14:40:32] frigate.app                    INFO    : Camera processor started for giardino: 495
2024-04-24 14:40:32.658990841  [2024-04-24 14:40:32] frigate.app                    INFO    : Camera processor started for salone: 496
2024-04-24 14:40:32.665787681  [2024-04-24 14:40:32] frigate.app                    INFO    : Capture process started for giardino: 497
2024-04-24 14:40:32.671435635  [2024-04-24 14:40:32] frigate.app                    INFO    : Capture process started for salone: 502
2024-04-24 14:40:35.257010238  [2024-04-24 14:40:35] frigate.comms.mqtt             ERROR   : Unable to publish to giardino/all: client is not connected
2024-04-24 14:40:35.257326396  [2024-04-24 14:40:35] frigate.comms.mqtt             ERROR   : Unable to publish to giardino/motion: client is not connected
2024-04-24 14:40:35.258455414  [2024-04-24 14:40:35] frigate.comms.mqtt             ERROR   : Unable to publish to giardino_interno/all: client is not connected
2024-04-24 14:40:35.258536063  [2024-04-24 14:40:35] frigate.comms.mqtt             ERROR   : Unable to publish to vialetto_esterno/all: client is not connected
2024-04-24 14:40:35.258604935  [2024-04-24 14:40:35] frigate.comms.mqtt             ERROR   : Unable to publish to balcone/all: client is not connected
2024-04-24 14:40:35.648920641  [2024-04-24 14:40:35] frigate.comms.mqtt             ERROR   : Unable to publish to salone/all: client is not connected
2024-04-24 14:40:35.649180239  [2024-04-24 14:40:35] frigate.comms.mqtt             ERROR   : Unable to publish to salone/motion: client is not connected
2024-04-24 14:41:34.699570752  [2024-04-24 14:41:34] frigate.comms.mqtt             ERROR   : Unable to publish to giardino/motion: client is not connected
2024-04-24 14:41:43.362561063  [2024-04-24 14:41:43] frigate.comms.mqtt             ERROR   : Unable to publish to stats: client is not connected
2024-04-24 14:41:57.631799707  [2024-04-24 14:41:57] frigate.comms.mqtt             ERROR   : Unable to publish to giardino/motion: client is not connected
2024-04-24 14:42:43.904554090  [2024-04-24 14:42:43] frigate.comms.mqtt             ERROR   : Unable to publish to stats: client is not connected
2024-04-24 14:43:44.447075947  [2024-04-24 14:43:44] frigate.comms.mqtt             ERROR   : Unable to publish to stats: client is not connected
2024-04-24 14:44:44.986375043  [2024-04-24 14:44:44] frigate.comms.mqtt             ERROR   : Unable to publish to stats: client is not connected
2024-04-24 14:45:45.565571027  [2024-04-24 14:45:45] frigate.comms.mqtt             ERROR   : Unable to publish to stats: client is not connected
2024-04-24 14:46:07.896044551  [2024-04-24 14:46:07] frigate.comms.mqtt             ERROR   : Unable to publish to giardino/motion: client is not connected
2024-04-24 14:46:23.854744303  [2024-04-24 14:46:23] frigate.comms.mqtt             ERROR   : Unable to publish to giardino/motion: client is not connected
2024-04-24 14:46:46.105557119  [2024-04-24 14:46:46] frigate.comms.mqtt             ERROR   : Unable to publish to stats: client is not connected
2024-04-24 14:47:46.644344673  [2024-04-24 14:47:46] frigate.comms.mqtt             ERROR   : Unable to publish to stats: client is not connected
2024-04-24 14:47:55.641340015  [2024-04-24 14:47:55] frigate.comms.mqtt             ERROR   : Unable to publish to giardino/motion: client is not connected
2024-04-24 14:48:47.190469340  [2024-04-24 14:48:47] frigate.comms.mqtt             ERROR   : Unable to publish to stats: client is not connected
2024-04-24 14:49:01.226477389  [2024-04-24 14:49:01] frigate.comms.mqtt             ERROR   : Unable to publish to giardino/motion: client is not connected
2024-04-24 14:49:47.725503208  [2024-04-24 14:49:47] frigate.comms.mqtt             ERROR   : Unable to publish to stats: client is not connected
2024-04-24 14:49:55.815571802  [2024-04-24 14:49:55] frigate.comms.mqtt             ERROR   : Unable to publish to giardino/motion: client is not connected
2024-04-24 14:50:13.236219231  [2024-04-24 14:50:13] frigate.comms.mqtt             ERROR   : Unable to publish to giardino/motion: client is not connected
2024-04-24 14:50:48.269628118  [2024-04-24 14:50:48] frigate.comms.mqtt             ERROR   : Unable to publish to stats: client is not connected`

**Comment by nerdosity** [comment]
Log from my mosquitto instance:

```
root@UNAS:~# cat /mnt/user/appdata/mosquitto/log/mosquitto.log 
1713962708: mosquitto version 2.0.18 starting
1713962708: Config loaded from /mosquitto-unraid/include_dir.conf.
1713962708: Opening ipv4 listen socket on port 1883.
1713962708: Opening ipv6 listen socket on port 1883.
1713962708: mosquitto version 2.0.18 running
1713962708: New connection from 192.168.0.144:55919 on port 1883.
1713962708: New client connected from 192.168.0.144:55919 as 3Pg7EITFEb35SB7ON0zE1C (p2, c1, k60, u'homeassistant').
1713962708: New connection from 192.168.1.208:47211 on port 1883.
1713962708: New client connected from 192.168.1.208:47211 as mosqsub|385-DAFANG (p1, c1, k60, u'homeassistant').
1713962709: New connection from 192.168.1.20:64635 on port 1883.
1713962709: New client connected from 192.168.1.20:64635 as 7060d64c9be349b98e986e92ce98a336 (p2, c1, k30, u'homeassistant').
1713962713: New connection from 192.168.2.5:50744 on port 1883.
1713962713: New client connected from 192.168.2.5:50744 as nukihub (p2, c0, k15, u'homeassistant').
1713962964: mosquitto version 2.0.18 terminating
1713962964: Saving in-memory database to /mosquitto/data/mosquitto.db.
1713962965: mosquitto version 2.0.18 starting
1713962965: Config loaded from /mosquitto-unraid/include_dir.conf.
1713962965: Opening ipv4 listen socket on port 1883.
1713962965: Opening ipv6 listen socket on port 1883.
1713962965: mosquitto version 2.0.18 running
1713962965: New connection from 192.168.1.20:63751 on port 1883.
1713962965: New client connected from 192.168.1.20:63751 as 7060d64c9be349b98e986e92ce98a336 (p2, c1, k30, u'homeassistant').
1713962965: New connection from 192.168.1.208:47212 on port 1883.
1713962965: New client connected from 192.168.1.208:47212 as mosqsub|385-DAFANG (p1, c1, k60, u'homeassistant').
1713962965: New connection from 192.168.0.144:50465 on port 1883.
1713962965: New client connected from 192.168.0.144:50465 as 3Pg7EITFEb35SB7ON0zE1C (p2, c1, k60, u'homeassistant').
1713962970: New connection from 192.168.2.5:50745 on port 1883.
1713962970: New client connected from 192.168.2.5:50745 as nukihub (p2, c0, k15, u'homeassistant').
```

**Comment by NickM-27** [comment]
more likely that it tries but can't reach it

**Comment by NickM-27** [comment]
what is the config?

**Comment by imanbakker** [comment]
are you also running it inside truenas scale? and updated to dragonfish today?
It took me a while but i am fairly certain the problem originates form on the truenass/frigate side in my case.
i tried deploying a new docker inside docker with an older version of frigate (truecharts) and that also doesn't work.

so i am guessing the combination Truenas scale: dragonfish and frigate isnt working properly on the network side but i cant figure out where it is going wrong.
unfortunatly i cannot rollback my truenas scale installation..

**Comment by nerdosity** [comment]
> are you also running it inside truenas scale? and updated to dragonfish today? It took me a while but i am fairly certain the problem originates form on the truenass/frigate side in my case. i tried deploying a new docker inside docker with an older version of frigate (truecharts) and that also doesn't work.
> 
> so i am guessing the combination Truenas scale: dragonfish and frigate isnt working properly on the network side but i cant figure out where it is going wrong. unfortunatly i cannot rollback my truenas scale installation..

nope, it's a unRAID custom machine.
https://hub.docker.com/r/cmccambridge/mosquitto-unraid/
and 
https://hub.docker.com/r/blakeblackshear/frigate/

**Comment by nerdosity** [comment]
![image](https://github.com/blakeblackshear/frigate/assets/26657281/f19f905f-3a9c-4d7e-84c7-19934a5ade5b)
I run this from the shell, and it's unable to connect to 192.168.0.1. 

Any suggestions?

**Comment by silvioped** [comment]
In my case, in the frigate configuration yaml, removing the :1883 in the "host: "  solved the problem.

**Comment by pypeaday** [comment]
I ran into this today but my issue was pretty simple - mqtt is running on home assistant, but my home assistant VM IP changed on reboot (silly me) so my frigate config was pointing to wrong IP. Simple problem but took me quite a while to diagnose

**Comment by mh-g** [comment]
I had the same issue (frigate dockerized, mosquitto not, both on the same raspi4 with debian 11, ipv4 and ipv6). My issue was that I used the hostname. Everything worked, after I changed to its ipv4 address. Not nice, but at least working.

**Comment by fluffyspace** [comment]
Clearly there's still an issue about this. My configuration starts with 

```mqtt:
  enabled: true
  host: 192.168.1.204
  port: 1883
  user: REDACTED
  password: REDACTED
  topic_prefix: frigate
  client_id: frigate
  stats_interval: 15```

and network_mode: host  in docker compose. Messages are only 
`frigate.comms.mqtt             DEBUG   : Unable to publish to REDACTED/all/active: client is not connected`

and mosquitto (which is not in docker but rather installed with apt-get on pc) logs show no sign of frigate connecting. Other subscribers and publishers are connecting successfully. All services are running on same server. So I can't figure it out, is there something wrong with docker? Any help is appreciated.

**Comment by fluffyspace** [comment]
Clearly there's still an issue with this. My configuration starts with 

```
mqtt:
  enabled: true
  host: 192.168.1.204
  port: 1883
  user: REDACTED
  password: REDACTED
  topic_prefix: frigate
  client_id: frigate
  stats_interval: 15
```
and `network_mode: host`  in docker compose. Messages are only 
`frigate.comms.mqtt             DEBUG   : Unable to publish to REDACTED/all/active: client is not connected`

and mosquitto (which is not in docker but rather installed with apt-get on pc) logs show no sign of frigate connecting. Other subscribers and publishers are connecting successfully. All services are running on same server. So I can't figure it out, is there something wrong with docker or frigate? Is Frigate optimized as Home Assistant add-on? Any help is appreciated.

**Comment by cjm571** [comment]
Had these same symptoms, and it turned out to be an issue with my reverse proxy (Traefik). I initially had Traefik listening on port 1883, with the configuration set up to pass that traffic through to the MQTT container. 

When I reconfigured my MQTT container to instead directly use the host's port 1883, I no longer saw MQTT disconnects, nor the "client is not connected" errors in Frigate.

**Comment by ejwight** [comment]
I'm running frigate and mosquitto mqtt broker in their own docker compose containers.  The frigate logs do not show any attempt to connect to mosquitto.  Mosquitto logs also do not show any such attempt to connect from Frigate.  I know that the mqtt broker is reachable because I can connect from home assistant and an mqtt utility.  My issue seems similar to what Nerdosity reported.

**Comment by stokedout** [comment]
> The frigate logs do not show any attempt to connect to mosquitto. Mosquitto logs also do not show any such attempt to connect from Frigate. I know that the mqtt broker is reachable because I can connect from home assistant and an mqtt utility. My issue seems similar to what Nerdosity reported.

It probably can't find the host but the timeout is quite long. Remember when services talk to each other within a compose container they can use different hosts and ports from what you use outside. So check your environment variables inside the compose file and make sure they reference the {service_name}:{port}.

**Comment by ejwight** [comment]
This is what I have in my config.  

mqtt:
  enabled: true
  host: 172.16.209.127 #this is the IP of my host machine where the containers are running, and what I use to reach the mqtt broker from other clients
  port: 1883
  user: mqtt-frigate
  password: xxxxxxxx

This is what is in my compose docker compose file:

services:
  frigate:
    container_name: frigate
    privileged: true 
    restart: unless-stopped
    stop_grace_period: 30s 
    image: ghcr.io/blakeblackshear/frigate:stable

    shm_size: "512mb" # update for your cameras based on calculation above
    devices:
      - /dev/bus/usb:/dev/bus/usb 
      - /dev/dri/renderD128:/dev/dri/renderD128 

    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /home/jeffw/frigate/config:/config
      - /mnt/roxyridge-data/frigate/media:/media/frigate
      - type: tmpfs 
        target: /tmp/cache
        tmpfs:
          size: 1000000000
    ports:
      - "8971:8971"
      - "5000:5000" # Internal unauthenticated access. Expose carefully.
      - "8554:8554" # RTSP feeds
      - "8555:8555/tcp" # WebRTC over tcp
      - "8555:8555/udp" # WebRTC over udp
    environment:
      FRIGATE_RTSP_PASSWORD: "xxxxxxxx"

ENV within the frigate container:

root@71732924eb1a:/opt/frigate# env
NVIDIA_VISIBLE_DEVICES=all
HOSTNAME=71732924eb1a
S6_CMD_WAIT_FOR_SERVICES_MAXTIME=0
PWD=/opt/frigate
NVIDIA_DRIVER_CAPABILITIES=compute,video,utility
TRANSFORMERS_NO_ADVISORY_WARNINGS=1
HOME=/root
TOKENIZERS_PARALLELISM=true
LANG=C.UTF-8
S6_LOGGING_SCRIPT=T 1 n0 s10000000 T
OPENCV_FFMPEG_LOGLEVEL=8
FRIGATE_RTSP_PASSWORD=xxxxxxxx
TERM=xterm-256color
SHLVL=1
LC_ALL=C.UTF-8
PATH=/usr/local/go2rtc/bin:/usr/local/tempio/bin:/usr/local/nginx/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
_=/usr/bin/env

**Comment by stokedout** [comment]
Assuming this is config from within docker compose/container:

```
mqtt:
enabled: true
host: 172.16.209.127 #this is the IP of my host machine where the containers are running, and what I use to reach the mqtt broker from other clients
port: 1883
user: mqtt-frigate
password: xxxxxxxx
```

Try changing `172.16.209.127` to `host.docker.internal` so that it points back to your host

**Comment by ejwight** [comment]
Thanks for the suggestion, the logs still don't show any attempt to connect.

s6-rc: info: service s6rc-fdholder: starting
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service s6rc-fdholder successfully started
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service log-prepare: starting
s6-rc: info: service log-prepare successfully started
s6-rc: info: service nginx-log: starting
s6-rc: info: service go2rtc-log: starting
s6-rc: info: service frigate-log: starting
s6-rc: info: service certsync-log: starting
s6-rc: info: service nginx-log successfully started
s6-rc: info: service go2rtc-log successfully started
s6-rc: info: service go2rtc: starting
s6-rc: info: service certsync-log successfully started
s6-rc: info: service frigate-log successfully started
s6-rc: info: service go2rtc successfully started
s6-rc: info: service go2rtc-healthcheck: starting
s6-rc: info: service frigate: starting
s6-rc: info: service go2rtc-healthcheck successfully started
s6-rc: info: service frigate successfully started
s6-rc: info: service nginx: starting
2025-12-16 08:39:01.859489661  [INFO] Preparing new go2rtc config...
2025-12-16 08:39:01.863004109  [INFO] Preparing Frigate...
2025-12-16 08:39:01.863968536  [INFO] Starting NGINX...
2025-12-16 08:39:01.958895658  [INFO] Starting Frigate...
2025-12-16 08:39:02.456049709  [INFO] Starting go2rtc...
2025-12-16 08:39:02.559345524  08:39:02.558 INF go2rtc platform=linux/amd64 revision=b2399f3 version=1.9.2
2025-12-16 08:39:02.559352422  08:39:02.558 INF config path=/dev/shm/go2rtc.yaml
2025-12-16 08:39:02.559354303  08:39:02.558 INF [rtsp] listen addr=:8554
2025-12-16 08:39:02.559356231  08:39:02.558 INF [webrtc] listen addr=:8555/tcp
2025-12-16 08:39:02.559636422  08:39:02.559 INF [api] listen addr=:1984
s6-rc: info: service nginx successfully started
s6-rc: info: service certsync: starting
s6-rc: info: service certsync successfully started
s6-rc: info: service legacy-services: starting
2025-12-16 08:39:02.961364007  [INFO] Starting certsync...
s6-rc: info: service legacy-services successfully started
2025-12-16 08:39:04.586003875  [2025-12-16 08:39:04] frigate.util.config            INFO    : Checking if frigate config needs migration...
2025-12-16 08:39:04.604153716  [2025-12-16 08:39:04] frigate.util.config            INFO    : frigate config does not need migration...
2025-12-16 08:39:04.625449832  [2025-12-16 08:39:04] frigate.app                    INFO    : Starting Frigate (0.15.0-cea210d)
2025-12-16 08:39:04.632173526  [2025-12-16 08:39:04] peewee_migrate.logs            INFO    : Starting migrations
2025-12-16 08:39:04.632357935  [2025-12-16 08:39:04] peewee_migrate.logs            INFO    : There is nothing to migrate
2025-12-16 08:39:04.658531437  [2025-12-16 08:39:04] frigate.app                    INFO    : Recording process started: 347
2025-12-16 08:39:04.658598144  [2025-12-16 08:39:04] frigate.app                    INFO    : Review process started: 353
2025-12-16 08:39:04.660806871  [2025-12-16 08:39:04] frigate.app                    INFO    : go2rtc process pid: 98
2025-12-16 08:39:04.672172919  [2025-12-16 08:39:04] detector.coral                 INFO    : Starting detection process: 363
2025-12-16 08:39:04.672516512  [2025-12-16 08:39:04] frigate.app                    INFO    : Embedding process started: 365
2025-12-16 08:39:04.685896088  [2025-12-16 08:39:04] frigate.app                    INFO    : Output process started: 388
2025-12-16 08:39:04.686081179  [2025-12-16 08:39:04] frigate.comms.mqtt             DEBUG   : Unable to publish to model_state: client is not connected

**Comment by ejwight** [comment]
I resolved my issue of frigate not connected to the mosquitto broker.  I did it by 1) creating a network in docker, 2) directing both frigate and mosquitto to connect to that network, then 3) referring to the mosquitto host by name in my frigate config file.

I suspect this may be unique to running both frigate and mosquitto in their own docker containers.  In my case, it was creating a unique network for each and they were not talking to each other.

1) docker network create -d bridge HomeAssistantNetwork

2) Frigate docker compose file (see the last 6 lines, I added the same lines to the mosquitto docker compose file).
services:

  frigate:
    container_name: frigate
 ....
    networks:
      - HomeAssistantNetwork  

networks:
  HomeAssistantNetwork:
    external: true

3) Frigate config file
mqtt:
  enabled: true
  host: mosquitto #name of the mosquitto container
  port: 1883
  user: mqtt-frigate
  password: xxxxxxxx

....
